src/Microsoft.AspNet.OData.Shared/Formatter/ODataInputFormatterHelper.cs

```diff
@@ -82,6 +82,15 @@ internal static class ODataInputFormatterHelper
                 oDataReaderSettings.Version = version;
                 oDataReaderSettings.MaxProtocolVersion = version;
 
+                ODataPath path = internalRequest.Context.Path;
+                ODataDeserializerContext readContext = getODataDeserializerContext();
+                readContext.Path = path;
+                readContext.Model = model;
+                readContext.ResourceType = type;
+                readContext.ResourceEdmType = expectedPayloadType;
+
+                oDataReaderSettings.EnablePropertyNameCaseInsensitive = !readContext.DisableCaseInsensitiveRequestPropertyBinding;
```

Comment:
Do you have test cases for both scenarios (when enabled and disabled)?

Reply:
Added the tests